tutorials/cloud-iot-rtdp/streaming/src/main/java/com/google/cloud/solutions/rtdp/Simulator.java

```diff
@@ -0,0 +1,162 @@
+import org.eclipse.paho.client.mqttv3.IMqttDeliveryToken;
+import org.eclipse.paho.client.mqttv3.MqttCallback;
+import org.eclipse.paho.client.mqttv3.MqttClient;
+import org.eclipse.paho.client.mqttv3.MqttConnectOptions;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.eclipse.paho.client.mqttv3.MqttMessage;
+import org.eclipse.paho.client.mqttv3.MqttPersistenceException;
+import org.eclipse.paho.client.mqttv3.persist.MemoryPersistence;
+import org.joda.time.DateTime;
+
+/*
+ * Simulator emulates multiple MQTT clients and establishes sessions to Cloud IoT Core.
+ * It randomly chooses MQTT sessions and sends a specified number of messages repeatedly.
+ */
+public class Simulator implements MqttCallback {
+  public static final int NUM_DEVICES = 30;
+
+  private static PrivateKey loadKeyFile(String filename, String algorithm) throws Exception {
+    byte[] keyBytes = Files.readAllBytes(Paths.get(filename));
+    PKCS8EncodedKeySpec spec = new PKCS8EncodedKeySpec(keyBytes);
+    KeyFactory kf = KeyFactory.getInstance(algorithm);
+    return kf.generatePrivate(spec);
+  }
+
+  private static String createJwt(String projectId, String privateKeyFile, String algorithm)
+      throws Exception {
+    DateTime now = new DateTime();
+    JwtBuilder jwtBuilder =
+        Jwts.builder()
+            .setIssuedAt(now.toDate())
+            .setExpiration(now.plusMinutes(60).toDate())
+            .setAudience(projectId);
+
+    if (algorithm.equals("RS256")) {
+      PrivateKey privateKey = loadKeyFile(privateKeyFile, "RSA");
+      return jwtBuilder.signWith(SignatureAlgorithm.RS256, privateKey).compact();
+    } else if (algorithm.equals("ES256")) {
+      PrivateKey privateKey = loadKeyFile(privateKeyFile, "EC");
+      return jwtBuilder.signWith(SignatureAlgorithm.ES256, privateKey).compact();
+    } else {
+      throw new IllegalArgumentException(
+          "Invalid algorithm " + algorithm + ". Should be one of 'RS256' or 'ES256'.");
+    }
+  }
+
+  private MqttClient client;
+  private String deviceId;
+  private String topic;
+  private Random r;
+  private String sz = "20";
+  private String lat;
+  private String lng;
+
+  public Simulator(int id, MqttOptions options) {
+    r = new Random();
+    lat = options.lat + String.format("%04d", r.nextInt(10000));
+    lng = options.lng + String.format("%04d", r.nextInt(10000));
+    deviceId = options.deviceId + Integer.toString(id);
+    String mqttServerAddress =
+        String.format("ssl://%s:%s", options.mqttBridgeHostname, options.mqttBridgePort);
+    String mqttClientId =
+        String.format(
+            "projects/%s/locations/%s/registries/%s/devices/%s",
+            options.projectId, options.cloudRegion, options.registryId, deviceId);
+    MqttConnectOptions connectOptions = new MqttConnectOptions();
+    connectOptions.setMqttVersion(MqttConnectOptions.MQTT_VERSION_3_1_1);
+    connectOptions.setUserName("unused");
+    try {
+      connectOptions.setPassword(
```

Comment:
Note that you will need to refresh the token after 60 minutes.  May be worth adding a comment or actually implementing as is done in this sample:

https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/iot/api-client/http_example/src/main/java/com/google/cloud/iot/examples/HttpExample.java#L168...L179